2015/presentations/with modern_peripherals: Python and Flask Piotr Dyba/text.md

```diff
@@ -0,0 +1,208 @@
+# with modern_peripherals: Python and Flask - Piotr Dyba
+
+##Introduction
+
+Auto-scrolling sites, glance-following ads, and gesture friendly web pages are 
+coming! Over the last few years three products emerged that enable interaction 
+with computer in a new way: Myo Armband, Leap Motion Controller and EyeTribe. 
+Using web sockets it is possible to interact using this new devices with web 
+apps and devices in completely new way. Lets start with really short 
+introduction to Web Sockets, after that I’ll get through implementation, 
+wrappers and an imaginative app for each of the new devices.
+
+##Web Sockets and Python socket solution
+
+In short, web sockets is a protocol providing full-duplex communication channels 
+over a single TCP connection available in web browsers 
+
+![Alt text](./001_web_sockets.png "01 Web Sockets Diagram")
+
+It starts with handshake which upgrades the HTTP connection to use the Web Socket protocol.
+After that there messages exchanges from and too server triggered by user 
+it self or depending on other circumstances defined by programmer – 
+that’s what interests us the most. At the end, one of the sides must send a kill 
+message, which will terminate the connection. 
+Build in Python library is nice but writing JS from scratch on front end would 
+be time consuming so I did not use it. The second solution was SocketIo and its 
+python extension would work well but it requires additional socket only server 
```

Comment:
How about: Because of that I decided to use the Socket.IO which enables real-time bidirectional event-based communication. It works on every platform, browser or device. It's a powerful solution but it requires additional socket server. I used Flask-SocketIO...